sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/django.yaml

```
@@ -0,0 +1,81 @@
+provider:
+  metadata:
+    name: Django
+    category: Application server
+    message: Make sure this Django key gets revoked, changed, and removed from the code.
+
+  rules:
+    - id: default-secret-key
+      metadata:
+        name: Django secret keys should not be disclosed
+      detection:
+        matching:
+          # https://github.com/django/django/blob/main/django/core/management/utils.py#L79
+          pattern: "\\b(django-insecure-[a-z0-9!@#$%^&*(-_=+)]{50})"
+        post:
+          statisticalFilter:
+            threshold: 4.9 # Based on significant sampling, 5.0 < entropy < 5.7
+      examples:
+        - text: >
+            KEY = 'django-insecure-r&lvybzry1*k+qq)=x-!=0yd5l5#1gxzk!82@ru25*ntos3_9^'
+          containsSecret: true
+          match: django-insecure-r&lvybzry1*k+qq)=x-!=0yd5l5#1gxzk!82@ru25*ntos3_9^
+
+    - id: settings.py-secret-key
+      metadata:
+        name: Django secret keys should not be disclosed
+      detection:
+        pre:
+          include:
+            paths:
+              - "*/settings.py"
+        matching:
+          pattern: "\\bSECRET_KEY(?:_FALLBACKS)?\\s*=.{0,15}?['\"]([^'\"]+)"
+          context:
+            patternAround: "(?i)\\bdjango\\b"
+        post:
+          patternNot: "(?i)(sample|example|x{4,}|0{4,})"
+      examples:
+        - text: >
+            # See https://docs.djangoproject.com/en/1.9/howto/deployment/checklist/
+            # SECURITY WARNING: keep the secret key used in production secret!
+            SECRET_KEY = 'r&lvybzry1*k+qq)=x-!=0yd5l5#1gxzk!82@ru25*ntos3_9^'
+
+          containsSecret: true
+          match: r&lvybzry1*k+qq)=x-!=0yd5l5#1gxzk!82@ru25*ntos3_9^
+
+    - id: env-secret-key
+      metadata:
+        name: Django secret keys should not be disclosed
+      detection:
+        pre:
+          include:
+            paths:
+              - "*.env"
+        matching:
+          # Here, matching string delimiters inside the secret is accepted to reduce the pattern complexity
+          pattern: "(?m)\\bSECRET_KEY(?:_FALLBACKS)?\\s*=\\s*(.*)$"
+          context:
+            matchEither:
+              - patternAround: "(?i)\\bdjango\\b"
+              - patternAround: "ALLOWED_HOSTS\\s*=" # Chosen because has the most chances of being configured per env
+        post:
+          heuristicFilter:
+            heuristics:
+              - "path"
+          patternNot: "(?i)({{.*?}}|sample|example|x{4,}|0{4,})"
```

The opening curly braces near the start of the pattern need to be escaped. It appears that the closing braces are ok, at least according to Regex101.

```suggestion
          patternNot: "(?i)(\\{\\{.*?}}|sample|example|x{4,}|0{4,})"
```